File: Theme_34_Lesson_6_CRUD_Application/Theme_34_Lesson_6_CRUD_Application/Student.cs


namespace Theme_34_Lesson_6_CRUD_Application
{
    public class Student
    {
        public int Id { get; set; }
        public string? Firstname { get; set; }
        public string? Lastname { get; set; }
        public string Age { get; set; }

        public Student()
        {

        }

        public Student(string? firstname, string? lastname, string age)
        {
            Firstname = firstname;
            Lastname = lastname;
            Age = age;
        }

        public string Display(Student student)
        {
            string stroke = $"Имя: {student.Firstname}, Фамилия: {student.Lastname}, Возраст: {student.Age}";
            return stroke;
        }
    }





}

File: Помелов М.А/NodeJS/Lesson29_Repositori/src/entities/Student.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, ManyToOne, Join } from 'typeorm';
import { Teacher } from './Teacher';

@Entity()
export class Student {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column()
    firstname!: string;
    @Column()
    lastname!: string;
    @Column()
    age!: number;

    @ManyToOne(() => Teacher, teacher => teacher.students)
    teachers!: Teacher[];
}